### ISSUE TITLE
Memory leak on Metal in triangle example

### ISSUE THREAD

**Issue by djsell** [gfx-rs/wgpu-native#132]
The triangle example has a memory leak on Metal.  Memory usage grows unbound and quite rapidly. ~5MB/second

I found this out because I was trying to use wgpu-native in a C# project.  My app showed the same behavior, so I verified against the example app.

![image](https://user-images.githubusercontent.com/182778/133151401-e8eb1909-ad7e-4545-a6cb-3a42aeb76f58.png)

wgpu-native: 5fe297fc1146abc8123ffca59c4e861c1487def4
os: 11.5.2
gpu: Intel HD Graphics 530 1536 MB

**Comment by djsell** [comment]
Hmmm.. Sorry, this may actually be an issue with running the app from Xcode.  Seems like there is some Metal debug interaction that causes the extra allocs. I don't know if that is normal or to be expected.  Dunno how people would normally debug their games with Xcode if that is how it typically works.

The app doesn't exhibit the memory leak when it is ran standalone.

I did notice that the app goes into the gigs of memory almost instantly once the window is occluded.
Related: https://github.com/gfx-rs/wgpu/issues/1783

**Comment by kvark** [comment]
right. We merged a fix for this, so the next wgpu-native update should pick it up.

**Comment by rajveermalviya** [comment]
@djsell can you check if this still occurs for you? If not I think this can be closed. Thanks.
